scripts/setup.py

```diff
@@ -49,47 +57,89 @@ def __init__(self,config,*args,**kwargs):
             self.b_prev.pack(side=LEFT,padx=40)
             self.b_next.pack(side=RIGHT,padx=40)
             self.widgets=[]
-            
+
             #variables
             #PAGE 0 email/upload
             self.want_email_var = IntVar()
             self.want_email_var.set(config.enable_email == True)
             def on_want_email_change(*args):
                 self.config.enable_email = self.want_email_var.get() != 0
             self.want_email_var.trace("w",on_want_email_change)
-            
+
             self.want_upload_var = IntVar()
             self.want_upload_var.set(config.enable_upload == True)
             def on_want_upload_change(*args):
                 self.config.enable_upload = self.want_upload_var.get() != 0
             self.want_upload_var.trace("w",on_want_upload_change)
-            
+
+
+            if self.printer_selection_enable == True:
+                self.want_print_var = IntVar()
+                self.want_print_var.set(config.enable_print == True)
+                def on_want_print_change(*args):
+                    self.config.enable_print = self.want_print_var.get() !=0
+                    if self.config.enable_print == True:
+                        self.use_print_list = Listbox(self.main_frame)
+                        self.use_print_list.bind('<<ListboxSelect>>', on_use_printer)
+                        #self.use_print_list.pack()
+                        self.__erase_page()
+                        self.widgets.pop(0)
+                        self.widgets.insert(0,[self.want_email_cb, self.want_upload_cb, self.want_print_cb,self.use_soft_keyboard_cb,self.use_print_list])
+                        conn = cups.Connection()
```

Comment:
This fails on my system because cups was not set up.
As soon as I check the printer feature, the setup screen goes blank and an error appears in the console (see screenshots below)

Simplest way to fix would be to try/catch this and in case of failure:
- Alert user that he/she must configure cups (possibly linking to a fix webpage)
- set the print option to false (and disabled)

![before_option_select](https://user-images.githubusercontent.com/16169826/43969976-2bb68e0a-9ccc-11e8-839c-5db68ef926d5.png)
![after_option_select](https://user-images.githubusercontent.com/16169826/43969980-30415e50-9ccc-11e8-84a4-718015ff14f3.png)